topics/functions-intro.md

```diff
@@ -0,0 +1,353 @@
+# Introduction to Redis Functions
+ 
+As of Redis 7.0, functions offer a new programmability approach.
+Functions have evolved from the [Eval Scripts](/topics/eval-intro) feature that was, in turn, added in version 2.6.
+
+You can skip the following prologue if you're new to Redis or want to jump right into the action.
+
+## Prologue (or what's wrong with Eval Scripts?)
+
+Prior versions of Redis made scripting available only via the `EVAL` command, which allows a Lua script to be sent for execution by the server.
+One of the core use cases for [Eval Scripts](/topics/eval-intro) is implementing and exposing a richer API composed of core Redis commands and workflow logic.
+Such APIs can perform conditional updates across multiple keys, possibly combining several different data types, to maintain a consistent view of entities (e.g., user profiles) somewhat like schemas in traditional databases.
+
+Using `EVAL` requires that the application sends the entire script for execution every time.
```

Comment:
maybe one of these two lines can be elaborated a bit (before going into the next lines that lists the limitations).
like stressing that Eval scripts are part of the client application (being executed on the server), and NOT an API enhancement. which is why they're not persisted or named.